packages/vite-plugin-cloudflare/src/node-js-compat.ts

```diff
@@ -76,65 +174,50 @@ export function injectGlobalCode(
  * But Vite only allows us to configure aliases at the shared options level, not per environment.
  * So instead we alias these to a virtual module, which are then handled with environment specific code in the `resolveId` handler
  */
-export function getNodeCompatAliases() {
+function getNodeCompatAliases() {
 	const aliases: Record<string, string> = {};
-	Object.keys(preset.alias).forEach((key) => {
+	Object.keys(env.alias).forEach((key) => {
 		// Don't create aliases for modules that are already marked as external
-		if (!preset.external.includes(key)) {
+		if (!env.external.includes(key)) {
 			aliases[key] = CLOUDFLARE_VIRTUAL_PREFIX + key;
 		}
 	});
 	return aliases;
 }
 
-/**
- * Attempt to resolve the `id` to an unenv alias or polyfill.
- */
-export function resolveNodeCompatId(
-	environment: Environment,
-	workerConfig: WorkerConfig,
-	id: string
-) {
-	const aliased = resolveNodeAliases(id, workerConfig) ?? id;
-
-	if (aliased.startsWith("unenv/")) {
-		const resolvedDep = require.resolve(aliased).replace(/\.cjs$/, ".mjs");
-		if (environment.mode === "dev" && environment.depsOptimizer) {
-			const dep = environment.depsOptimizer.registerMissingImport(
-				aliased,
-				resolvedDep
-			);
-			return dep.id;
-		} else {
-			return resolvedDep;
-		}
-	}
-}
-
 /**
  * Get an array of modules that should be considered external.
  */
-export function getNodeCompatExternals(): string[] {
-	return preset.external;
+function getNodeCompatExternals(): string[] {
+	return env.external;
 }
 
 /**
  * Convert any virtual module Id that was generated by the aliases returned from `getNodeCompatAliases()`
  * back to real a module Id and whether it is an external (built-in) package or not.
  */
-function resolveNodeAliases(source: string, workerConfig: WorkerConfig) {
-	if (
-		!source.startsWith(CLOUDFLARE_VIRTUAL_PREFIX) ||
-		!isNodeCompat(workerConfig)
-	) {
+function dealiasVirtualNodeJSImports(
+	source: string,
+	workerConfig: WorkerConfig
+) {
+	if (!source.startsWith(CLOUDFLARE_VIRTUAL_PREFIX)) {
 		return;
 	}
 
 	const from = source.slice(CLOUDFLARE_VIRTUAL_PREFIX.length);
-	const alias = preset.alias[from];
+	if (!isNodeCompat(workerConfig)) {
+		// We are not in node compat mode so just return the original module specifier
+		return from;
+	}
 
-	if (alias && preset.external.includes(alias)) {
-		throw new Error(`Alias to external: ${source} -> ${alias}`);
+	const alias = env.alias[from];
+	if (!alias) {
+		return;
```

Comment:
What's actually happened if we get here and `alias` is `undefined`? If this an error and should it be an assertion?

Reply:
Right I think this is an error state.

Reply:
3b726f727e59f14f69c2dc098dc60737340e298f